### PR TITLE
change private ip logic as per rajiv

### PR DESCRIPTION
**Pull request by vivekjhauipath**

### CODE REVIEW

**Review by adeepsri — DISMISSED** [review]
LGTM

**Review by adeepsri — APPROVED** [review]
LGTM

File: platform/onebox/backup_and_restore/ceph/get-credentials.sh [hunk 1]
```diff
@@ -5,11 +5,24 @@ This scipt will generate json file for creds to be used with import export
...
 [Script Version -> 21.4]
 '
+
+readonly PRIVATE_IP=$1
+
 function initialize_variables() {
-	# Gets private ip of machine so that it can be connected within the VM
-	OBJECT_GATEWAY_EXTERNAL_HOST=$(hostname -i)
+	if [ -z "$PRIVATE_IP" ]; then
+		# Gets private ip of machine so that it can be connected within the VM
```
Comment (mohitseth-nitrr) [inline platform/onebox/backup_and_restore/ceph/get-credentials.sh:16]: nit: indentation

File: platform/onebox/backup_and_restore/ceph/get-credentials.sh [hunk 2]
```diff
@@ -5,11 +5,24 @@ This scipt will generate json file for creds to be used with import export
...
+		# Seems to be set as localhost on some customer machines
+		#OBJECT_GATEWAY_EXTERNAL_HOST=$(hostname -i)
+		PRIVATE_ADDRESS=$(ip -o route get to 8.8.8.8 | sed -n 's/.*src \([0-9.]\+\).*/\1/p')
+    	#This is needed on k8s 1.18.x as $PRIVATE_ADDRESS is found to have a newline
+    	OBJECT_GATEWAY_EXTERNAL_HOST=$(echo "$PRIVATE_ADDRESS" | tr -d '\n')
+    else
+    	OBJECT_GATEWAY_EXTERNAL_HOST=$PRIVATE_IP
```
Comment (mohitseth-nitrr) [inline platform/onebox/backup_and_restore/ceph/get-credentials.sh:23]: nit: lets put an echo statement here saying Host is set to this

**Review by mohitseth-nitrr — APPROVED** [review]
Minor changes, otherwise LGTM

File: platform/onebox/backup_and_restore/ceph/get-credentials.sh (hunk 1 above)
Comment (vivekjhauipath) [reply platform/onebox/backup_and_restore/ceph/get-credentials.sh:16]: github is being weird again

File: platform/onebox/backup_and_restore/ceph/get-credentials.sh (hunk 2 above)
Comment (vivekjhauipath) [reply platform/onebox/backup_and_restore/ceph/get-credentials.sh:23]: makes sense, will do later, basically echo what user passed and what was set

**Pull request merged**